src/dataflow/src/render/context.rs

```diff
@@ -91,8 +114,42 @@ where
     S::Timestamp: Lattice + Refines<T>,
 {
     /// Creates a new empty Context.
-    pub fn new() -> Self {
+    pub fn for_dataflow(dataflow: &DataflowDesc) -> Self {
+        let as_of_frontier = dataflow
+            .as_of
+            .clone()
+            .unwrap_or_else(|| Antichain::from_elem(0));
+
+        // TODO (materialize#1720): replace `first_export_id` by some form of
+        // dataflow identifier.
```

Comment:
Ah, we do have an issue, but later on!

Reply:
Is this comment related to the code that follows it, or just after the `assert!` ?

Reply:
Good point. I reordered this confusingly. Should be better now.